bin/node/cli/src/service.rs

```diff
@@ -552,12 +553,17 @@ pub fn new_full_base(
 }
 
 /// Builds a new service for a full client.
-pub fn new_full(
-	config: Configuration,
-	disable_hardware_benchmarks: bool,
-) -> Result<TaskManager, ServiceError> {
-	new_full_base(config, disable_hardware_benchmarks, |_, _| ())
-		.map(|NewFullBase { task_manager, .. }| task_manager)
+pub fn new_full(config: Configuration, cli: Cli) -> Result<TaskManager, ServiceError> {
+	let database_source = config.database.clone();
+	let task_manager = new_full_base(config, cli.no_hardware_benchmarks, |_, _| ())
+		.map(|NewFullBase { task_manager, .. }| task_manager);
+
+	storage_monitor::StorageMonitorService::try_spawn(
+		cli.storage_monitor,
+		database_source,
+		&task_manager.as_ref().unwrap().spawn_essential_handle(),
+	)?;
+	task_manager
```

Comment:
This is purely subjective and is really a nitpick, but the control flow seems inverted to me. What about something like:

```rust
    ...
    let task_manager = new_full_base(...)?.task_manager;

    task_manager.spawn_essential_handle().spawn(
        "storage-monitor",
        None,
        StorageMonitorService::try_run(cli.storage_monitor, database_source)?,
    );

    Ok(task_manager)
}
```

Reply:
For me both approaches are fine. Please also refer to: https://github.com/paritytech/substrate/pull/13082#discussion_r1067035779